src/aind_behavior_curriculum/behavior.py

```diff
@@ -0,0 +1,50 @@
+"""
+Base Behavior Models
+"""
+
+from pydantic import BaseModel, ConfigDict, Field
+
+
+class AindBehaviorModel(BaseModel):
+    """
+    Defines Pydantic configurations applied to all behavior models.
+    BaseModel: Validate arguments on initialization.
+    Configurations:
+        - extra='forbid':
+            Do not allow a model to be initalized with undocumented parameters.
+        - validate_assignment=True:
+            Revalidate fields against schema on any change to model instance.
+        - validate_defaults=True:
+            Validate default fields on subclasses.
+        - strict=True:
+            Enforce strict typing.
+    """
+
+    model_config = ConfigDict(
+        extra="forbid",
+        validate_assignment=True,
+        validate_defaults=True,
+        strict=True,
+    )
+
+
+class Task(AindBehaviorModel):
+    """
+    Set of parameters associated with a mouse task.
+    Task parameters may be updated and are revalidated on assignment.
+    """
+
+    name: str = Field(..., description="Name of the task.", frozen=True)
```

Comment:
Should we add `task_version` and `task_repo` here? @bruno-f-cruz

Reply:
Sure, I am not sure what kind of feedback @jwong-nd is looking for here. Should we also start flagging missing properties? I think just going with what I had in the example repo will cover most cases (?).

Reply:
Feel free to comment on any properties you feel should be included-- you guys have been involved in the design process and have thought about this much more than me.

Reply:
For the core Task I think it is really just the version. This property is used to version the schema/task model.
Do you have an opinion on these two implementations?

1) Using SemVer. I think this is important as it gives users the ability to quickly perform version range checks and even bumps automatically without too much hassle. We also agreed to adhere strickly to SemVer.
https://github.com/AllenNeuralDynamics/Aind.Behavior.MouseUniversity/blob/1ea05884ac58276c72ab01bcc069f0de8c62e281/src/aind_behavior_mouse_university/base/__init__.py#L37

2) Duplicating the version information as a ClassVar. This is a QOL change that would allow users to query the version of the schema without having to instantiate the model. (You could in theory force a non-validated model, but if feels clunky...)
https://github.com/AllenNeuralDynamics/aind-data-schema/discussions/738

Reply:
@bruno-f-cruz what if we used `aind_behavior_curriculum.__version__` for this property? Would reduce complexity here.